main2.py

```diff
@@ -0,0 +1,38 @@
+from flask import Flask, render_template, request, redirect, url_for, make_response
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def input_text():
+    return render_template('index.html')
+
+
+@app.route('/set_cookie', methods=['POST'])
+def set_cookie():
+    username = request.form['username']
+    email = request.form['email']
+
+    response = make_response(redirect(url_for('greet')))
+    response.set_cookie('username', username)
+    response.set_cookie('email', email)
+    return response
+
+
+@app.route('/greet')
+def greet():
+    username = request.cookies.get('username')
+    email = request.cookies.get('email')
+    return render_template('greet.html', username=username, email=email)
+
+
+@app.route('/delite', methods=['POST'])
+def delite():
+    response = make_response(redirect(url_for('input_text')))
+    response.delete_cookie('username')
+    response.delete_cookie('email')
+    return response
```

Comment:
Вообще у вас как-то многовато функций представления. Можно же проверять тип запроса и уместить все в 2 функции
```suggestion
@app.route('/', methods=['GET', 'POST'])
def input_text():
    if request.method == 'POST':
        username = request.form['username']
        email = request.form['email']

        response = make_response(redirect(url_for('greet')))
        response.set_cookie('username', username)
        response.set_cookie('email', email)
        return response

    return render_template('index.html')


@app.route('/greet', methods=['GET', 'POST'])
def greet():
    if request.method == 'POST':
        response = make_response(redirect(url_for('input_text')))
        response.delete_cookie('username')
        response.delete_cookie('email')
        return response

    username = request.cookies.get('username')
    email = request.cookies.get('email')
    return render_template('greet.html', username=username, email=email)
```